src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,62 @@
+package duke.task;
+
+/**
+ * This class is used to create tasks.
+ * Each Task must contain a description and can either be marked done or not done.
+ */
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    protected static final String ICON_DONE = "[X]";
+    protected static final String ICON_NOT_DONE =  "[ ]";
+    public static final String TODO_ACRONYM = "T";
+    protected static final String TODO_LOGO = "[" + TODO_ACRONYM + "]";
+    public static final String DEADLINE_ACRONYM = "D";
+    protected static final String DEADLINE_LOGO = "[" + DEADLINE_ACRONYM + "]";
+    public static final String EVENT_ACRONYM = "E";
+    protected static final String EVENT_LOGO = "[" + EVENT_ACRONYM + "]";
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public String getDescription() {
+        return this.description;
+    }
+
+    public String getStatusIcon() {
+        return (this.isDone ? ICON_DONE : ICON_NOT_DONE); //marks task done with "X"
+    }
+
+    public String markAsDone() {
+        if (!this.isDone) {
+            this.isDone = true;
+            return "Well done! I've marked this task as done. ^_^";
+        } else {
```

Comment:
The `else` may not be needed since there is already a `return` statement inside the `if` clause.